### PR TITLE
AWS RDS cluster Table is added

### PR DESCRIPTION
**Pull request by laukik-uptycs**
By submitting a PR to this repository, you agree to the terms within the [Code of Conduct](https://github.com/Uptycs/cloudquery/blob/master/CODE_OF_CONDUCT.md). Please see the [contributing guidelines](https://github.com/Uptycs/cloudquery/blob/master/CONTRIBUTING.md) for how to create and submit a PR for this repo.

### Description

Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

Include any links supporting this change such as:

- GitHub Issue/PR number addressed or fixed
- Related pull requests/issues from other repos

### Testing

Describe the testing done. Setup required for testing. Details of unit or other tests added as a part of the PR.

### CODE REVIEW

File: extension/aws/rds/aws_rds_cluster.go [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+	paginator := rds.NewDescribeDBClustersPaginator(svc, params)
+
+	for {
+		page, err := paginator.NextPage(osqCtx)
+		if err != nil {
+			utilities.GetLogger().WithFields(log.Fields{
+				"tableName": "aws_rds_cluster",
+				"account":   accountId,
+				"region":    *region.RegionName,
+				"task":      "ListCertificates",
```
Comment (alende-uptycs) [inline extension/aws/rds/aws_rds_cluster.go:250]: DescribeDBClusters

File: extension/aws/rds/aws_rds_cluster.go [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+			}).Error("failed to process region")
+			return resultMap, err
+		}
+		byteArr, err := json.Marshal(page)
+		if err != nil {
+			utilities.GetLogger().WithFields(log.Fields{
+				"tableName": "aws_rds_cluster",
+				"account":   accountId,
+				"region":    *region.RegionName,
+				"task":      "ListCertificates",
```
Comment (alende-uptycs) [inline extension/aws/rds/aws_rds_cluster.go:261]: DescribeDBClusters

File: extension/aws/rds/table_config.json [hunk 3]
```diff
@@ -0,0 +1,1002 @@
...
+        {
+          "sourceName": "DBClusters_ActivityStreamStatus",
+          "targetName": "activity_stream_status",
+          "targetType": "TEXT",
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_AllocatedStorage",
+          "targetName": "allocated_storage",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:37]: table.BigIntColumn in go file instead of TextColumn

File: extension/aws/rds/table_config.json [hunk 4]
```diff
@@ -0,0 +1,1002 @@
...
+        {
+          "sourceName": "DBClusters_AvailabilityZones",
+          "targetName": "availability_zones",
+          "targetType": "TEXT",
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_BacktrackConsumedChangeRecords",
+          "targetName": "backtrack_consumed_change_records",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:73]: table.BigIntColumn in go file instead of TextColumn

File: extension/aws/rds/table_config.json [hunk 5]
```diff
@@ -0,0 +1,1002 @@
...
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_BacktrackWindow",
+          "targetName": "backtrack_window",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:79]: ****

File: extension/aws/rds/table_config.json [hunk 6]
```diff
@@ -0,0 +1,1002 @@
...
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_BackupRetentionPeriod",
+          "targetName": "backup_retention_period",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:85]: ****

File: extension/aws/rds/table_config.json [hunk 7]
```diff
@@ -0,0 +1,1002 @@
...
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_Capacity",
+          "targetName": "capacity",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:91]: table.BigIntColumn in go file instead of TextColumn

File: extension/aws/rds/table_config.json [hunk 8]
```diff
@@ -0,0 +1,1002 @@
...
+        {
+          "sourceName": "DBClusters_ClusterCreateTime_loc_cacheZone_name",
+          "targetName": "cluster_create_time_loc_cache_zone_name",
+          "targetType": "TEXT",
+          "enabled": false
+        },
+        {
+          "sourceName": "DBClusters_ClusterCreateTime_loc_cacheZone_offset",
+          "targetName": "cluster_create_time_loc_cache_zone_offset",
+          "targetType": "INTEGER",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:157]: table.IntegerColumn in go file instead of TextColumn

File: extension/aws/rds/table_config.json [hunk 9]
```diff
@@ -0,0 +1,1002 @@
...
+        {
+          "sourceName": "DBClusters_ClusterCreateTime",
+          "targetName": "cluster_create_time",
+          "targetType": "TEXT",
+          "enabled": true
+        },
+        {
+          "sourceName": "DBClusters_ClusterCreateTime_ext",
+          "targetName": "cluster_create_time_ext",
+          "targetType": "BIGINT",
```
Comment (alende-uptycs) [inline extension/aws/rds/table_config.json:115]: Change the column type in go file for all BIGINT attributes

File: utilities/utilities.go [hunk 10]
```diff
@@ -49,6 +49,7 @@ func ReadTableConfig(jsonEncoded []byte) error {
 		config.initParsedAttributeConfigMap()
 		TableConfigurationMap[tableName] = config
 	}
+
```
Comment (alende-uptycs) [inline utilities/utilities.go:52]: let's unstage this unrequire file change

**Comment by alende-uptycs** [comment]
Also, update `Dockerfile` and `extension/aws/tables.md`

**Comment by laukik-uptycs** [comment]
**Description**

I have added the _AWS RDS Cluster_ information 

1. Added aws_rds_cluster.go file 
2. Created the necessary table-config.json file and enabled all first-level attributes
3. added those enabled columns in the table
4. Added the table name in table.md file
5. Added RDS table info in the docker file

**References**
https://docs.aws.amazon.com/cli/latest/reference/rds/describe-db-clusters.html

**Testing**
Tested using OSQuery

**Comment by laukik-uptycs** [comment]
I have done the necessary changes please review and do the needful

File: extension/aws/rds/aws_rds_cluster.go [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+		table.TextColumn("clone_group_id"),
+		table.TextColumn("cluster_create_time"),
+		// table.BigIntColumn("cluster_create_time_ext"),
+		// table.TextColumn("cluster_create_time_loc"),
+		// table.BigIntColumn("cluster_create_time_loc_cache_end"),
+		// table.BigIntColumn("cluster_create_time_loc_cache_start"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_is_dst"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_name"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_offset"),
```
Comment (alende-uptycs) [inline extension/aws/rds/aws_rds_cluster.go:44]: Change the integer column data types as well wherever needed

**Review by alende-uptycs — APPROVED** [review]

**Pull request merged**